core/src/main/scala/zio/config/Config.scala

```diff
@@ -129,7 +129,7 @@ object Config {
    *
    * {{{
    *    final case class kafkaConfig(server: String, serde: String)
-   *    nested("KAFKA")(string("SERVERS") |@| string("SERDE"))(KafkaConfig.apply, KafkaConfig.unapply)
+   *    nested("KAFKA")(string.at("SERVERS") |@| string.at("SERDE"))(KafkaConfig.apply, KafkaConfig.unapply)
```

Comment:
Need some tests/proofs. Isn't it?
nested("KAFKA")(string.at("SERVERS"))  == string.at("SERVERS").at("KAFKA")

Reply:
> @senia-psm also could you add sbt runAllExamples as part of the build. We don’t need breaking examples. If there are any examples without assertions let’s change that as well :)

I'm getting `Nonzero exit code returned from runner: 1` without any details on `sbt runAllExamples`

Reply:
All tests are already reimplemented with `.at`. I have no idea what should I test here.

Reply:
The idea is, now users have two ways in which they form `nested` paths, unlike before.

So let's add a simple test, which is
```scala
nested("KAFKA")(string.at("SERVERS")) == string.at("SERVERS").at("KAFKA")


```
also worth testing
```scala
list(string.at("path")) = list(string).at("path")
```

Reply:
The reason is, currently we have two ways in which some one can list. Conceptually, this was not the case before. The only way to create a list, is , for instance:

```sala
list(string("path"))
```

This is why I think, worth adding a test

Reply:
Actually no, right now `list(string.at("path")) = list(string).at("path")`, but I'm going to break it in #261 since there is a bug related to it. I mentioned this bug earlier in discord.
I'm going to enforce this kind of behavior to fix #261:
`list(string).at("path")` <-> `{path: ["a"]}`
`list(string.at("path"))` <-> `[{path: "a"}]`